Sources/UInt128.swift

```diff
@@ -339,30 +300,21 @@ extension UInt128: UnsignedInteger {
         self.init(value.toUIntMax())
     }
     public func toUIntMax() -> UIntMax {
-        return UIntMax(value.lowerBits)
+        return lo.toUIntMax()
     }
     // MARK: Hashable Conformance
     public var hashValue: Int {
-        return Int(
-            value.lowerBits.hashValue ^ value.upperBits.hashValue
-        )
-    }
-    // MARK: ForwardIndexType Conformance
-    public func successor() -> UInt128 {
-        return self &+ 1
-    }
-    // MARK: BidirectionalIndexType Conformance
-    public func predecessor() -> UInt128 {
-        return self &- 1
+        return lo.hashValue ^ hi.hashValue
```

Comment:
👎 

As the result needs to be an Int, I would prefer to keep the current behaviour to cast it to an Int before returning to make this more explicit.